crates/opencascade/src/primitives.rs

```diff
@@ -213,9 +213,28 @@ impl AsRef<Wire> for Wire {
 impl Wire {
     pub fn from_edges<'a>(edges: impl IntoIterator<Item = &'a Edge>) -> Self {
         let mut make_wire = ffi::BRepBuilderAPI_MakeWire_ctor();
+        let mut edges = edges.into_iter().collect::<Vec<_>>();
+
+        // This will get updated by missing edges and exit when all edges are added
+        while edges.len() > 0 {
+            let mut missing_edges = Vec::with_capacity(edges.len());
+            let start_len = edges.len();
+
+            for edge in edges {
+                make_wire.pin_mut().add_edge(&edge.inner);
+                // If MakeWire IsDone returns false it means the edge was not added due to no shared vertices
+                // We will assume it is due to the edges being out of order and will attempt to add again
+                if make_wire.pin_mut().IsDone() != true {
```

Comment:
Ah I was close, it's `consolidateWires` in cadquery, which indeed calls [`ConnectEdgesToWires`](https://dev.opencascade.org/doc/refman/html/class_shape_analysis___free_bounds.html#a4809fe812b6a663a286df71b4d7bda3b)

Reply:
I also think this is quite nice to have in the `from_edges` - not a panic but a `Result` if all the edges are not added.